enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,340 @@
+* As an Operator developer, I want to have a standard framework to define tokenized authentication requirements and
+  consume them, per supported cloud, so that my operator will work on token-based authentication clusters.
+* As an Operator Hub browser, I want to know which operators support tokenized cloud auth and on which clouds so I can
+  see only a filtered list of operators that will work on the given cluster.
+* As an Operator Hub browser, I want to be informed / reminded in the UI that the cluster only supports TAT 
+  authentication with the Cloud Provider, so that I don't confuse the cluster with one that will try to mint long lived
+  credentials.
+* As an Operator Hub browser, I want to be able to easily provide what's required to the OLM operators I install through
+  the UI.
+* As the HyperShift team, where CCO is not installed so the only supported authentication mode is via TAT, I want the
+  Red Hat branded operators that must reach the Cloud Provider API, to be enabled to work with TAT credentials in a
+  consistent, and automated fashion so that customer can use those operators as easily as possible, driving the use of
+  layered products.
+
+### Goals
+
+Allow Day-2 Operators to access cloud provider resources as seamlessly as possible when being installed, used and
+updated on STS enabled clusters.
+
+While providing the above goal, allow for multi-tenancy. In this sense multi-tenancy means that an operator may
+enable its operands to communicate with the cloud provider instead of communicating with the cloud itself. Operands 
+would use the same set of credentials as the operator. The operator will need to maintain its own logic to minimize 
+conflicts when sharing credentials with operands.
+
+Operator authors have a way to notify, guide, and assist Day-2 Operator admins in providing the required cloud provider
+credentials matched to their permission needs for install and update.
+
+Ideally, a solution here will work in both HyperShift (TAT always) and non-HyperShift clusters.
+
+### Non-Goals
+
+Day 1 operators are not included in this proposal, they are pre-provisioned for TAT enabled clusters (at least 
+for HyperShift)
+
+Bring Your Own Credentials (BYOC) where an operator can manage and distribute credentials to its operands are out of
+scope.
+
+
+## Proposal
+
+**Cloud Credential Operator (CCO) changes**: Adds a token-aware mode, while nominally set in "Manual" mode, CCO will
+look for and process CredentialRequests referenced in Operator CRs when:
+
+```bash
+# infrastructure platform is AWS
+$ oc get infrastructure cluster -o jsonpath --template '{ .status.platform }'
+AWS
+
+# credentialsMode is "Manual"
+$ oc get cloudcredential cluster -o jsonpath --template '{ .spec.credentialsMode }'
+Manual
+
+# serviceAccountIssuer is non empty
+$ oc get authentication cluster -o jsonpath --template='{ .spec.serviceAccountIssuer }'
+abutcher-oidc.s3.us-east-1.amazonaws.com
+```
+
```

Comment:
Why is this an implicit as opposed to an explicit mechanism?

Reply:
The original explicit mechanism proposed was a new `credentialsMode` which entails installing the cluster in that mode and I didn't think a new `credentialsMode` made sense given that the existing `credentialsMode` of `"Manual"` is still relevant as the credentials are manually maintained by the cluster admin. One really needs to check the above to infer STS* capability but it would be cool if there was a singular explicit way to know.

Reply:
\+1 for more obvious detection if an operator should create a CredentialsRequest or not. Right now, our operator does not create it in `Manual` mode, with this enhancement, we would need to check `infrastructure`, `cloudcredential` and `authentication`. That looks IMO quite convoluted.

Reply:
Perhaps we should externalize the function that does the detection in CCO?

Reply:
Okay we covered the reasoning for not exposing `util.IsTimedTokenCluster()` in the API offline. @abutcher if you still feel opposed, could you put notes here about why not?

Reply:
Not opposed, would be fine with adding a util to CCO that does the check if it will be useful to consumers.

Reply:
In our sync yesterday we talked about the possibility of setting a configv1 field but decided that the installer wouldn't know it was making a timed token enabled cluster, only that the the installer would be blindly applying manifests that cause the cluster to end up as one.

Reply:
ah, ah thanks for de-addling me ;-) Okay marking this resolved and adding as a possibility to the text here.